src/config.c

```diff
@@ -229,17 +229,87 @@ CONFIG_GETTER(getMinOperationWorkers) {
   return sdscatprintf(ss, "%lu", config->minOperationWorkers);
 }
 
-// MT_MODE, WORKER_THREADS
-CONFIG_SETTER(trySetDeprecatedWorkersConfig) {
+/************************************ DEPRECATION CANDIDATES *************************************/
+
+enum MTMode {
+  MT_MODE_OFF,
+  MT_MODE_ONLY_ON_OPERATIONS,
+  MT_MODE_FULL,
+};
+
+// Old configuration
+enum MTMode mt_mode_config = MT_MODE_OFF;
+size_t numWorkerThreads_config = 0;
+
+// WORKER_THREADS
+CONFIG_SETTER(setDeprWorkThreads) {
   RedisModule_Log(RSDummyContext, "warning", "MT_MODE and WORKER_THREADS are deprecated, use WORKERS and MIN_OPERATION_WORKERS instead");
-  int acrc = AC_Advance(ac); // Consume the value argument
-  RETURN_STATUS(acrc);
+  size_t newNumThreads;
+  int acrc = AC_GetSize(ac, &newNumThreads, AC_F_GE0);
+  CHECK_RETURN_PARSE_ERROR(acrc);
+  if (newNumThreads > MAX_WORKER_THREADS) {
+    QueryError_SetErrorFmt(status, QUERY_ELIMIT, "Number of worker threads cannot exceed %d", MAX_WORKER_THREADS);
+    return REDISMODULE_ERR;
```

Comment:
Didn't we had a test for that?

Reply:
We still have for the new configuration. We don’t have bad input tests for immutable configs